code/__DEFINES/_protect.dm

```diff
@@ -11,9 +10,19 @@
 ##Path/CanProcCall(procname){\
     return FALSE;\
 }\
+##Path/Read(savefile/savefile){\
+	del(src);\
+}\
+##Path/Write(savefile/savefile){\
+	return;\
+}\
 ##Path/can_vv_mark(){\
 	return FALSE;\
 }
+#else
+#define VV_PROTECT(Path)
+#endif
+// we del instead of qdel because for security reasons we must ensure the datum does not exist if Read is called. qdel will not enforce this.
 
 /**
  * Makes a path read-only to view variables.
```

Comment:
readonly probably needs /Read() override too

Reply:
kinda usless since you can still tag (and potentially) json_encode to get its value